### PR TITLE
Lifting useLoadTableColumnsToCache out of Flyouts

### PR DESCRIPTION
**Pull request by sejli**
### Description
* Flyouts had an issue where `startLoading` would poll indefinitely when the flyout gets closed. Lifting the `useLoadTableColumnsToCache` hook out of flyouts stops this behavior. 
* Updated UI on empty associated objects table

### Issues Resolved
N/A

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/datasources/components/manage/accelerations/create_accelerations_flyout/create/create_acceleration.tsx [hunk 1]
```diff
@@ -41,16 +40,23 @@ import { hasError } from './utils';
...
 export const CreateAcceleration = ({
   selectedDatasource,
   resetFlyout,
+  loadStatus,
+  startLoading,
+  stopLoading,
   databaseName,
   tableName,
 }: CreateAccelerationProps) => {
+  console.log(loadStatus);
```
Comment (ps48) [inline public/components/datasources/components/manage/accelerations/create_accelerations_flyout/create/create_acceleration.tsx:59]: Let's remove console.log

File: public/components/datasources/components/manage/accelerations/acceleration_table.tsx [hunk 2]
```diff
@@ -68,6 +69,7 @@ export const AccelerationTable = ({
     actionType: null,
     selectedItem: null,
   });
+  const { loadStatus, startLoading, stopLoading } = useLoadTableColumnsToCache();
```
Comment (ps48) [inline public/components/datasources/components/manage/accelerations/acceleration_table.tsx:72]: should this be declared in dataconnections page? What happens when we close flyout and switch the tab to associated objects

File: public/components/datasources/components/manage/accelerations/acceleration_table.tsx [hunk 3]
```diff
@@ -168,6 +170,9 @@ export const AccelerationTable = ({
                 <CreateAccelerationFlyoutButton
                   dataSourceName={dataSourceName}
                   renderCreateAccelerationFlyout={renderCreateAccelerationFlyout}
+                  loadStatus={loadStatus}
+                  startLoading={startLoading}
+                  stopLoading={stopLoading}
```
Comment (ps48) [inline public/components/datasources/components/manage/accelerations/acceleration_table.tsx:175]: Why are we only passing, hooks to load table fields? Create acceleration flyout uses:
1. Load databases: https://github.com/opensearch-project/dashboards-observability/blob/ce9db203e4922d0b3e83e5ce6f09b1e4c9f23a66/public/components/datasources/components/manage/accelerations/create_accelerations_flyout/selectors/selector_helpers/load_databases.tsx#L41
2. Load tables/accelerations: https://github.com/opensearch-project/dashboards-observability/blob/ce9db203e4922d0b3e83e5ce6f09b1e4c9f23a66/public/components/datasources/components/manage/accelerations/create_accelerations_flyout/selectors/selector_helpers/load_objects.tsx#L48-L57
3. Load table Columns: which is edited here. 
4. Direct query Hook: https://github.com/opensearch-project/dashboards-observability/blob/ce9db203e4922d0b3e83e5ce6f09b1e4c9f23a66/public/components/datasources/components/manage/accelerations/create_accelerations_flyout/create/create_acceleration_button.tsx#L30
Should all of the above lifted up in state?

File: public/components/datasources/components/manage/associated_objects/associated_objects_tab.tsx [hunk 4]
```diff
@@ -72,8 +72,17 @@ export const AssociatedObjectsTab: React.FC<AssociatedObjectsTabProps> = (props)
...
+    startLoadingTableColumns,
+    stopLoadingTableColumns,
   } = cacheLoadingHooks;
 
+  // const {
+  //   loadStatus: tableColumnsLoadStatus,
+  //   startLoading: startLoadingTableColumns,
+  //   stopLoading: stopLoadingTableColumns,
+  // } = useLoadTableColumnsToCache();
+
```
Comment (ps48) [inline public/components/datasources/components/manage/associated_objects/associated_objects_tab.tsx:85]: Can remove comments

File: public/components/datasources/components/manage/accelerations/acceleration_table.tsx (hunk 2 above)
Comment (sejli) [reply public/components/datasources/components/manage/accelerations/acceleration_table.tsx:72]: Associated objects tab also initializes the hook and passes it into the `Create Acceleration` flyout. The main issue was that when the flyout unmounts, we lose the reference to `stopLoading`, so the polling in `startLoading` won't stop. Lifting it out to the nearest parent component ensures that `stopLoading` still exists after the flyout unmounts and we can safely stop the loading.

File: public/components/datasources/components/manage/accelerations/acceleration_table.tsx (hunk 3 above)
Comment (sejli) [reply public/components/datasources/components/manage/accelerations/acceleration_table.tsx:175]: Hmm, that makes sense. Then, I can make the change such that the [giant hook](https://github.com/opensearch-project/dashboards-observability/blob/main/public/components/datasources/components/manage/data_connection.tsx#L76-L83) containing all the hooks (I'll add `useDirectQuery` as well) gets passed in as a prop. I can also create a type for `cacheLoadingHooks`.

Comment (ps48) [reply public/components/datasources/components/manage/accelerations/acceleration_table.tsx:175]: Ideally, we should use `Redux` or `useContext` to pass down element deep in tree. If that takes time we can use the giant hook. For the Giant hook, we'll have to expose direct query in public/types to be re-used in workbench

File: public/components/datasources/components/manage/associated_objects/associated_objects_details_flyout.tsx [hunk 5]
```diff
@@ -242,6 +259,10 @@ export const AssociatedObjectsDetailsFlyout = ({
     );
   }, [tableColumns]);
 
+  useEffect(() => {
+    stopLoading();
```
Comment (mengweieric) [inline public/components/datasources/components/manage/associated_objects/associated_objects_details_flyout.tsx:263]: not sure why stop loading every time component initializes but if you are looking for stop last loading you could do

`useEffect(() => {
  return () => stopLoading(); // stop pulling when unmounts
}, [])`

**Review by mengweieric — APPROVED** [review]

**Review by Swiddis — APPROVED** [review]

**Comment by sejli** [comment]
Fixes with #1598

**Pull request closed without merging**